concurrency/conc2_packed_task.cpp

```diff
@@ -0,0 +1,87 @@
+
+#include <vector>
+#include <string>
+#include <filesystem>
+#include <iostream>
+#include <future>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+
+std::vector<char> readFileContent(std::filesystem::path file_name);
+
+int main(int argc, char *argv[]) {
+
+    if (argc < 2) {
+        std::cerr << "please provide in the command line the file name" << std::endl;
+
+        return 1;
+    }
+    std::filesystem::path file_name = argv[1];
+
+    std::packaged_task<std::vector<char>(std::filesystem::path)> read_file_content(readFileContent);
+    std::future<std::vector<char>> file_content = read_file_content.get_future();
+    
+    try {
+        std::thread server_thread(std::move(read_file_content), file_name);
+        if (server_thread.joinable()) {
+            server_thread.join();
+        }
+    } catch (std::system_error& e) {
+        throw std::runtime_error(std::string("failed creating a thread") + e.what());
+    }
+    
+    try {
+        std::vector<char> data = file_content.get();
+
+        for (unsigned long int i = 0; i < data.size(); ++i) {
+            std::cout << data[i];
+        }
+        std::cout << std::endl;
+        
+    } catch (std::exception& e) {
+        
+        std::cerr << "read file content thread failed: " << e.what() << std::endl;
+    }
+
+
+
+    return 0;
+}
+
+std::vector<char> readFileContent(std::filesystem::path file_name) {
+
+    //throw std::runtime_error("Test throwing an error in the readFileContent thread"); // for testing
```

Comment:
remove commented out code

Reply:
this line if for testing (i throw on purpose to see if i get the behavior i expect upon failure)

Reply:
and yet

Reply:
and yet

Reply:
done